access/ms-teams/msapi/bot_framework_client.go

```diff
@@ -0,0 +1,135 @@
+package msapi
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gravitational/trace"
+)
+
+const (
+	botFrameworkAuthScope     = "https://api.botframework.com/.default"
+	botFrameworkBaseURL       = "https://smba.trafficmanager.net"
+	botFrameworkDefaultRegion = "uk"
+	botFrameworkVersion       = "v3"
+)
+
+// BotFrameworkClient represents client to MS Graph API
+type BotFrameworkClient struct {
+	Client
+}
+
+// ChatMember represents member of a chat
+type ChatMember struct {
+	ID                string `json:"id"`
+	ObjectID          string `json:"objectId"`
+	Name              string `json:"name"`
+	Email             string `json:"email"`
+	UserPrincipalName string `json:"userPrincipalName"`
+	TenantID          string `json:"tenantId"`
+	UserRole          string `json:"userRole"`
+}
+
+// PostActivityResponse represents json response with a single id field
+type PostActivityResponse struct {
+	ID string `json:"id"`
+}
+
+// botError represents MS Graph error
+type botError struct {
+	E struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error"`
+}
+
+// Error returns error string
+func (e *botError) Error() string {
+	return e.E.Code + " " + e.E.Message
+}
+
+// NewBotFrameworkClient creates MS Graph API client
+func NewBotFrameworkClient(config Config) *BotFrameworkClient {
+	region := config.Region
+	if region == "" {
+		region = botFrameworkDefaultRegion
+	}
+
+	baseURL := config.url.botFrameworkBaseURL
+	if baseURL == "" {
+		baseURL = botFrameworkBaseURL
+	}
+
+	baseURL = baseURL + "/" + region + "/" + botFrameworkVersion
```

Comment:
You can use the `CheckAndSetDefaults` pattern

Reply:
I guess, `CheckAndSetDefaults` here is applicable to the region only. Will move it to the `msapi.Config#CheckAndSetDefaults`.